src/shell.zig

```diff
@@ -891,45 +891,84 @@ pub const HttpOptions = struct {
         }
     };
 
-    content_type: ?ContentType,
-    authorization: ?[]const u8,
+    method: union(enum) { get, post: []const u8 },
+
+    content_type: ?ContentType = null,
+    authorization: ?[]const u8 = null,
+
+    response_body_size_max: u32 = 512 * stdx.KiB,
 };
 
-/// Issues an HTTP POST request with the given `body` to the given `url`.
+/// Issues an HTTP request to the given `url` and returns the response.
 ///
+/// The returned body is owned by the shell arena and doesn't need to be freed.
 /// If the response is not 200 OK, the response body is logged and an error is returned.
-pub fn http_post(shell: *Shell, url: []const u8, body: []const u8, options: HttpOptions) !void {
-    errdefer |err| log.err("failed to HTTP post to \"{s}\": {s}", .{ url, @errorName(err) });
+pub fn http_request(shell: *Shell, url: []const u8, options: HttpOptions) ![]const u8 {
```

Comment:
It's great to have _implementation_ be a single function, but let's have a

```
pub fn http_get
pub fn http_post
```

as the public API. 

At call site, we _always_ know statically whether we are getting or posting, we won't have a case where we'd want to dynamically switch between get and post. So, it would be a better interface if we statically encode function signature (`body` parameter), rather than type